core/domain/collection_domain.py

```diff
@@ -598,6 +751,25 @@ def delete_node(self, exploration_id):
                 exploration_id)
         del self.nodes[node_index]
 
+    def add_skill(self, skill_name):
+        """Adds the new skill domain object with the specified name."""
+
+        for _, skill in self.skills.iteritems():
+            if skill.name == skill_name:
+                raise ValueError(
```

Comment:
Should we do this validation as part of ``validate()`` instead? Ditto for ``delete_skill``